src/twisted/protocols/amp.py

```diff
@@ -674,11 +675,29 @@ def serialize(self):
                 raise TypeError("Unicode value for key %r not allowed: %r" % (k, v))
             if len(k) > MAX_KEY_LENGTH:
                 raise TooLong(True, True, k, None)
-            if len(v) > MAX_VALUE_LENGTH:
+            if len(v) > MAX_VALUE_LENGTH and not version2:
                 raise TooLong(False, True, v, k)
-            for kv in k, v:
-                w(pack("!H", len(kv)))
-                w(kv)
+
+            w(pack("!H", len(k)))
+            w(k)
+
+            if version2:
+                vio = BytesIO(v)
+                del v
+
+                # If the value is an exact multiple of 65535, the last
```

Comment:
how is the value "an exact multiple" of 65535 if it is not 65535 itself?

Reply:
Guess I should remove the word "exact" here.